src/main/java/com/googlecode/objectify/Key.java

```diff
@@ -246,11 +250,15 @@ public String toWebSafeString() {
 	 * <p>If no @Entity annotation is found, just uses the simplename as is.</p>
 	 */
 	public static String getKind(Class<?> clazz) {
-		String kind = getKindRecursive(clazz);
-		if (kind == null)
-			return clazz.getSimpleName();
-		else
-			return kind;
+		String kind = kindCache.get(clazz);

```

Comment:
The map is thread safe, but in theory you can compute the cache value more than once. `Map.computeIfAbsent()` would solve this.

Reply:
Java 7. No such method here.